nucleus/admin/monitor/src/main/java/fish/payara/admin/monitor/cli/ResetMonitoringLevels.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package fish.payara.admin.monitor.cli;
+
+import com.sun.enterprise.config.serverbeans.Config;
+import com.sun.enterprise.config.serverbeans.MonitoringService;
+import com.sun.enterprise.util.ColumnFormatter;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.inject.Inject;
+import org.glassfish.api.ActionReport;
+import org.glassfish.api.Param;
+import org.glassfish.api.admin.AdminCommand;
+import org.glassfish.api.admin.AdminCommandContext;
+import org.glassfish.api.admin.ExecuteOn;
+import org.glassfish.api.admin.RuntimeType;
+import org.glassfish.config.support.CommandTarget;
+import org.glassfish.config.support.TargetType;
+import org.glassfish.hk2.api.PerLookup;
+import org.glassfish.internal.api.Target;
+import org.jvnet.hk2.annotations.Service;
+import org.jvnet.hk2.config.ConfigSupport;
+import org.jvnet.hk2.config.TransactionFailure;
+
+/**
+ *
+ * @author Alan Roth
+ */
+@Service(name = "reset-monitoring-levels")
+@PerLookup
+@ExecuteOn({RuntimeType.DAS, RuntimeType.INSTANCE})
+@TargetType({CommandTarget.DAS, CommandTarget.STANDALONE_INSTANCE, CommandTarget.CLUSTER, CommandTarget.CONFIG})
+public class ResetMonitoringLevels implements AdminCommand {
+
+    @Param(name = "target", optional = true, defaultValue = "server-config")
+    String target;
+
+    @Param(name = "verbose", optional = true, shortName = "v")
+    private boolean verbose;
+
+    @Inject
+    private Target targetUtil;
+
+    @Inject
+    private Logger logger;
+
+    private MonitoringService monitoringService;
+
+    @Override
+    public void execute(AdminCommandContext context) {
+        ActionReport actionReport = context.getActionReport();
+
+        List<String> validModuleList = Arrays.asList(Constants.validModuleNames);
+        Config config = targetUtil.getConfig(target);
+        if (config != null) {
+            monitoringService = config.getMonitoringService();
+        } else {
+            actionReport.setMessage("Cound not find target: " + target);
+            actionReport.setActionExitCode(ActionReport.ExitCode.FAILURE);
+            return;
+        }
+
+        Map<String, String> enabledModules = getPreviouslyEnabledModules(validModuleList);
+
+        //Setting all modules to "OFF"
+        for (String module : validModuleList) {
```

Comment:
Do you need to set the ones that are already off to off? Does that actually do anything (e.g. poke it)?
Might be a slight performance improvement if you used your enabledModules map like you do when re-enabling them if it doesn't actually do anything.

Reply:
I had the same thought and assumed it did, since the jira asks to turn all monitoring items to OFF once. Since the reset command would be used when "AMX metrics are broken due to some reason" there may be factors I don't know about affecting them that setting to OFF could maybe fix, so I played it safe.

Reply:
However I'm happy to change it if you believe its not neccesery

Reply:
I don't think it's necessary - it's hard for an "off" metric to be wrong :P